File: backend/src/main/java/com/punarmilan/service/AdminReportService.java
package com.punarmilan.service;

import com.punarmilan.dto.ReportDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminReportService {
    Page<ReportDTO> getAllReports(Pageable pageable);

    void resolveReport(Long reportId, String adminNote);

    void dismissReport(Long reportId);

    void createReport(ReportDTO reportDTO, Long reporterId);
}

File: punarmilan-frontend/src/admin/services/adminReportService.js
import adminApi from './adminApi';

const adminReportService = {
    getAllReports: async (params) => {
        const response = await adminApi.get('/reports', { params });
        return response.data;
    },

    resolveReport: async (id, adminNote) => {
        const response = await adminApi.post(`/reports/${id}/resolve`, { adminNote });
        return response.data;
    },

    dismissReport: async (id) => {
        const response = await adminApi.post(`/reports/${id}/dismiss`);
        return response.data;
    }
};

export default adminReportService;
